src/components/common/FilterComponent.jsx

```diff
@@ -24,97 +21,68 @@ class FilterComponent extends React.Component {
+  handleCheckboxChange = (event) => {
+    const { option } = this.props;
+    const { checked, value } = event.target;
 
-    for (const key in checkedFilters) {
-      if (name === key) {
-        if (checkedFilters[name].has(label)) {
-          checkedFilters[name].delete(label);
-        } else {
-          checkedFilters[name].add(label);
-        }
-      }
-    }
-  };
+    const selection = {
+      label: value,
+      isChecked: checked
+    };
 
-  handleFilter = () => {
-    const { checkedFilters } = this.state;
-    const filters = {};
-
-    for (const key in checkedFilters) {
-      if (checkedFilters.hasOwnProperty(key)) {
-        filters[key] = [Array.from(checkedFilters[key])];
-      }
-    }
-
-    this.props.filterAction(
-      this.props.activePage,
-      this.props.limit,
-      filters
-    );
+    this.props.filterSelection(selection, option.title);
   };
 
   render() {
-    const { toggleOn, options } = this.props;
+    const { index, option } = this.props;
     const { activeIndex } = this.state;
 
-    if (!toggleOn) {
-      return null;
+    if (isEmpty(option)) {
+      return <span>Loading filters</span>;
     }
 
     return (
       <React.Fragment>
-        <Accordion as={Menu} vertical className="filter-menu">
-          {
-            !isEmpty(options)
-              ? options.map((option, index) => (
-                <Menu.Item key={uuidv4()}>
-                  <Accordion.Title
-                    active={activeIndex === index}
-                    content={option.title}
-                    index={index}
-                    onClick={this.handleTitleClick}
-                  />
-
-                  <Accordion.Content active={activeIndex === index}>
-                    <Form>
-                      {
-                        option.content.map(opt =>
-                          (<CheckboxComponent
-                            key={uuidv4()}
-                            label={opt.option}
-                            name={option.title}
-                            handleCheckboxChange={this.toggleCheckbox}
-                          />)
-                        )
-                      }
-                    </Form>
-                  </Accordion.Content>
-                </Menu.Item>
-                  ))
-            : <span>Loading filters...</span>
-          }
-        </Accordion>
-
-        <ArtButton
-          customCss="apply-filter"
-          buttonName="apply filters"
-          color="primary"
-          handleClick={this.handleFilter}
-        />
+        <Menu.Item>
+          <Accordion.Title
+            index={index}
+            active={activeIndex === index}
+            content={option.title}
+            onClick={this.handleTitleClick}
+          />
+          <Accordion.Content active={activeIndex === index}>
+            <Form>
+              {
+                  option.content.map((opt) => {
+                    const isChecked = this.props.selected[option.title]
```

Comment:
The ternary is unnecessary. `includes` always return a boolean value. Read more [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes#Return_value). Therefore change to:
```
const isChecked = this.props.selected[option.title].includes(opt.option);

Reply:
The ternary is necessary because when the component first loads, the `selected` state is an empty object. So the property that `isChecked` is looking for will be `undefined`. @sachungo

Reply:
And that is why `isChecked` has a default value of `false` in [this comment](https://github.com/AndelaOSP/art-dashboard/pull/137#discussion_r219035855). if `isChecked` is undefined it will never get passed to the checkbox component as a prop, and the default value will be used